src/org/jetbrains/java/decompiler/modules/decompiler/stats/Statement.java

```diff
@@ -981,6 +960,14 @@ public void setCopied(boolean copied) {
     this.copied = copied;
   }
 
+  public boolean isPhantom() {
+    return phantom;
+  }
+
+  public void setPhantom(boolean phantom) {
+    this.phantom = phantom;
+  }
+
```

Comment:
What about the nested statements in a phantom statement?

Reply:
I'm not sure, we need to work through the logic behind the ramifications of any statement potentially being phantom